c_glib/arrow-glib/file-system.cpp

```diff
@@ -0,0 +1,1050 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <arrow-glib/error.hpp>
+#include <arrow-glib/file-system.hpp>
+#include <arrow-glib/input-stream.hpp>
+#include <arrow-glib/output-stream.hpp>
+
+G_BEGIN_DECLS
+
+/**
+ * SECTION: file-system
+ * @section_id: file-system-classes
+ * @title: File system classes
+ * @include: arrow-glib/arrow-glib.h
+ *
+ * #GArrowFileStats is a class for a stats of file system entry.
+ *
+ * #GArrowFileSelector is a class for a selector for filesystem APIs.
+ *
+ * #GArrowFileSystem is an interface for file system.
+ *
+ * #GArrowSubTreeFileSystem is a delegator to another file system that is
+ * a logical view of a subtree of a file system, such as a directory in
+ * a local file system.
+ *
+ * #GArrowSlowFileSystem is a delegator to another file system.
+ * This inserts latencies at various points.
+ *
+ * #GArrowLocalFileSystem is a class for an implementation of a file system
+ * that accesses files on the local machine.
+ */
+
+/* arrow::fs::FileStats */
+
+typedef struct GArrowFileStatsPrivate_ {
+  std::shared_ptr<arrow::fs::FileStats> file_stats;
```

Comment:
How about removing `std::shared_ptr`?
It seems that `arrow::fs::FileStats` isn't shared.